src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -242,7 +242,10 @@ public override BoundNode VisitLambda(BoundLambda node)
             try
             {
                 _factory.CurrentFunction = node.Symbol;
-                return base.VisitLambda(node);
+                var visited = base.VisitLambda(node);
+                if (visited is BoundLambda lambda)
+                    visited = RewriteNullChecking(lambda);
+                return visited;
             }
```

Comment:
>} [](start = 12, length = 1)

Would it make sense to inline `base.VisitLambda()` here and avoid the new `RewriteNullChecking()` overload and additional `node.Update()`?
```C#
UnboundLambda unboundLambda = node.UnboundLambda;
BoundBlock body = (BoundBlock)Visit(node.Body);
body = RewriteNullChecking(body);
TypeSymbol type = this.VisitType(node.Type);
return node.Update(unboundLambda, node.Symbol, body, node.Diagnostics, node.Binder, type);
```
We could do something similar for `VisitLocalFunctionStatement` below.